module/Company/src/Company/Mapper/Label.php

```diff
@@ -0,0 +1,155 @@
+<?php
+
+namespace Company\Mapper;
+
+use Company\Model\JobLabel as LabelModel;
+use Doctrine\ORM\EntityManager;
+
+/**
+ * Mappers for labels.
+ *
+ */
+class Label
+{
+    /**
+     * Doctrine entity manager.
+     *
+     * @var EntityManager
+     */
+    protected $em;
+
+    /**
+     * Constructor.
+     *
+     * @param EntityManager $em
+     */
+    public function __construct(EntityManager $em)
+    {
+        $this->em = $em;
+    }
+
+    /**
+     * Saves all labels
+     *
+     */
+    public function save()
+    {
+        $this->em->flush();
+    }
+
+    /**
+     * Finds the label with the given id
+     *
+     * @param integer $packageID
+     */
+    public function findLabel($labelSlug)
+    {
+        return $this->getRepository()->findOneBy(['slug' => $labelSlug]);
+    }
+
+    public function findVisibleLabelByLanguage($labelLanguage)
+    {
+        $objectRepository = $this->getRepository(); // From clause is integrated in this statement
+        $qb = $objectRepository->createQueryBuilder('c');
+        $qb->select('c')->where('c.language=:lang');
+        $qb->andWhere('c.hidden=:hidden');
+        $qb->setParameter('lang', $labelLanguage);
+        $qb->setParameter('hidden', false);
+        $labels = $qb->getQuery()->getResult();
+
+        return $labels;
+    }
+
+    public function createNullLabel($lang, $translator)
+    {
+        $labelForJobsWithoutLabel =  new LabelModel();
+        $labelForJobsWithoutLabel->setHidden(false);
+        $labelForJobsWithoutLabel->setLanguageNeutralId(null);
+        $labelForJobsWithoutLabel->setLanguage($lang);
+        $labelForJobsWithoutLabel->setSlug("jobs");
+        $labelForJobsWithoutLabel->setName($translator->translate("Job"));
+
+        return $labelForJobsWithoutLabel;
+    }
+
+    /**
+     * Find the same label, but in the given language
+     *
+     */
+    public function siblingLabel($label, $lang)
+    {
+        $objectRepository = $this->getRepository(); // From clause is integrated in this statement
+        $qb = $objectRepository->createQueryBuilder('c');
+        $qb->select('c')->where('c.languageNeutralId=:labelID')->andWhere('c.language=:language');
+        $qb->setParameter('labelID', $label->getLanguageNeutralId());
+        $qb->setParameter('language', $lang);
+        $labels = $qb->getQuery()->getResult();
+        return $labels[0];
+    }
+    /**
+     * Inserts a new package into the given company
+     *
+     */
+    public function insert($lang, $id, $label = null)
+    {
+        if ($label == null) {
+            $label = new LabelModel();
+        }
+        $label->setLanguage($lang);
+        $label->setLanguageNeutralId($id);
+        $label->setHidden(false);
+        $this->em->persist($label);
+        $this->em->flush();
+        if ($id == -1) {
+            $id = $label->getId();
+        }
+        $label->setLanguageNeutralId($id);
+
+        return $label;
+    }
```

Comment:
Unless you have very good reasons not to, this is usually a job left for the service. You would create a label in the service, and then send it over to the mapper's `persist` method.

Reply:
This was consistent with the existing code within the module, but I strongly agree with your comment. I have changed it for the existing code as well.